Add option to export secrets as environment variables.

codiophile opened the pull request:
<!--
Thank you for proposing a pull request! Please note that SOME TESTS WILL
LIKELY FAIL due to how GitHub exposes secrets in Pull Requests from forks.
Someone from the team will review your Pull Request and respond.

Please describe your change and any implementation details below.
-->

**This adds an option to export the secrets as environment variables.**

We need this, because we are migrating from using our own custom action for fetching secrets and our old action would directly export the secrets as environment variables, so to be able to maintain API compatibility, we need this action to be able to do the same. Otherwise we would have to make significant changes to our workflows to enable the migration.

For others, this will be convenient, as most secrets you fetch will end up in an environment variable anyway. This removes the need for the manual step of putting the output from this action as an environment variable in the step where it is used.

As per my implementation, the feature is disabled by default and needs to be enabled to be used, which means that it is fully backwards compatible and will not affect users that don't want to use this feature.

I've tested it in one of our pipelines and it is working fine. I've also run linting and tests locally, without any errors or failures.

sethvargo commented on src/main.ts:
```diff
@@ -32,6 +39,9 @@ async function run(): Promise<void> {
     const minMaskLength = parseInt(getInput('min_mask_length'));
 
+    // Get the setting for whether to export secrets as environment variables.
+    const setEnv = getBooleanInput('set_env');
```
We prefer to use `parseBoolean` from google-github-actions/actions-utils; `getBooleanInput` has some pretty nasty edge cases.

sethvargo commented on src/main.ts:
```diff
@@ -54,6 +64,10 @@ async function run(): Promise<void> {
       });
 
       setOutput(ref.output, value);
+
+      if (setEnv) {
+        exportVariable(ref.output, value);
+      }
```
Since there's some branching logic here, we may want to add `logDebug` messages when the output is being set and when the variable is being exported.

codiophile replied:
It is unclear what you are asking for here. I searched for `logDebug` in the project, but got no results. It doesn't seem essential to me, but if you think it is necessary, could you please clarify, or perhaps add it yourself?

sethvargo commented on action.yml:
```diff
@@ -34,6 +34,13 @@ inputs:
     required: false
     default: '4'
 
+  set_env:
```
I think `export_environment` or `export_to_environment` might be a better name.

sethvargo reviewed (commented):
Hi @codiophile are you able to sign the Google CLA please? Also, we need to update the documentation with this new input. I would also prefer if the documentation clarified that the environment variable key is *exactly the same* as the output key.

This change seems mostly fine, but I am worried about feature sprawl. For example, I can see someone wanting to "uppercase" the secret names before exporting them, but I think we can handle those when they arise.

codiophile commented:
@sethvargo 

Thank you for your review. I signed the CLA and addressed the review comments, except the one about `logDebug`, as I'm not sure what you are after.
